plugins/optimization-detective/detect.js

```diff
@@ -487,10 +500,16 @@ export default async function detect( {
 			/** @type {Extension} */
 			const extension = await import( extensionModuleUrl );
 			extensions.set( extensionModuleUrl, extension );
+
+			const extensionLogger = createLogger( isDebug );
+			extensionLogger.prefix =
+				extension.consoleLogPrefix || '[Extension]';
+
 			// TODO: There should to be a way to pass additional args into the module. Perhaps extensionModuleUrls should be a mapping of URLs to args.
 			if ( extension.initialize instanceof Function ) {
 				const initializePromise = extension.initialize( {
 					isDebug,
+					logger: extensionLogger,
```

Comment:
Instead of passing the `logger` object, this could instead just pass the functions in the logger:

```suggestion
					...extensionLogger,
```

Reply:
Done in 73ebb74.